ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/Deduplicate.java

```diff
@@ -111,23 +110,15 @@ protected PubsubMessage processElement(PubsubMessage element) throws DuplicateId
    * {@link PTransform} to mark messages as seen, so any duplicates are removed.
    */
   public static class MarkAsSeen extends Deduplicate {
-    Integer ttlSeconds;
-    final ValueProvider<Duration> ttl;
+    final ValueProvider<Long> ttlSeconds;
 
-    int getTtlSeconds() {
-      if (ttlSeconds == null) {
-        ttlSeconds = (int) ttl.get().getStandardSeconds();
-      }
-      return ttlSeconds;
-    }
-
-    private MarkAsSeen(ValueProvider<URI> uri, ValueProvider<Duration> ttl) {
+    private MarkAsSeen(ValueProvider<URI> uri, ValueProvider<Long> ttlSeconds) {
       super(uri);
-      this.ttl = ttl;
+      this.ttlSeconds = ttlSeconds;
     }
 
     private String setex(byte[] id) {
-      return getJedis().setex(id, getTtlSeconds(), new byte[0]);
+      return getJedis().setex(id, ttlSeconds.get().intValue(), new byte[0]);
```

Comment:
Instead of putting off the `.intValue()` until here, I think we should use `ValueProvider<Integer>`, unless that also requires a method call, in which case this is fine

Reply:
I was using Long because java.time.Duration stores seconds as a long, so it seemed more natural to allow that precision.

Thinking more, though, I agree with you that it makes sense to narrow to `int` as soon as possible, since the whole point of this value is to pass it to the jedis API which expects and int.

I've left the `parseSeconds` method as returning a long, but now use a lambda inside `parseDecoderOptions` to cast to int and throw an exception on first access if the provided value overflows integer.

In general, it seems like good practice to do as much validation within `parseDecoderOptions` as possible, so that the parsed options are as safe and natural to use elsewhere as possible.